website/static/js/pages/profile-settings-addons-page.js

```diff
@@ -0,0 +1,186 @@
+            window.location.reload();
+        });
+        request.fail(function() {
+            var msg = 'Sorry, we had trouble saving your settings. If this persists please contact <a href="mailto: support@osf.io">support@osf.io</a>';
+            bootbox.alert({title: 'Request failed', message: msg});
+        });
+    };
+
+    if(unchecked.length > 0) {
+        var uncheckedText = $.map(unchecked, function(el){
+            return ['<li>', $(el).closest('label').text().trim(), '</li>'].join('');
+        }).join('');
+        uncheckedText = ['<ul>', uncheckedText, '</ul>'].join('');
+        bootbox.confirm({
+            title: 'Are you sure you want to remove the add-ons you have deselected? ',
+            message: uncheckedText,
+            callback: function(result) {
+                if (result) {
+                    submit();
+                } else{
+                    unchecked.each(function(i, el){ $(el).prop('checked', true); });
+                }
+            }
+        });
+    }
+    else {
+        submit();
+    }
+    return false;
+});
+
+var addonEnabledSettings = window.contextVars.addonEnabledSettings;
+for (var i=0; i < addonEnabledSettings.length; i++) {
+       var addonName = addonEnabledSettings[i];
+       if (typeof window.contextVars.addonsWithNodes !== 'undefined' && addonName in window.contextVars.addonsWithNodes) {
+           AddonPermissionsTable.init(window.contextVars.addonsWithNodes[addonName].shortName,
+                                      window.contextVars.addonsWithNodes[addonName].fullName);
+   }
+}
+
+/***************
+* OAuth addons *
+****************/
+var ConnectedProject = function(data) {
+    var self = this;
+    self.title = data.title;
+    self.id = data.id;
+    self.urls = data.urls;
+};
+
+var ExternalAccount = function(data) {
+    var self = this;
+    self.name = data.display_name;
+    self.id = data.id;
+
+    self.connectedNodes = ko.observableArray();
+
+    ko.utils.arrayMap(data.nodes, function(item) {
+        self.connectedNodes.push(new ConnectedProject(item));
+    });
+
+    self.deauthorizeNode = function(node) {
+        console.log(node);
+        var url = node.urls.deauthorize;
+        $.ajax({
+            url: url,
+            type: 'DELETE'
+        }).done(function(data) {
+            self.connectedNodes.remove(node);
+        }).fail(function(xhr, status, error) {
+            Raven.captureMessage('Error deauthorizing node: ' + node.id, {
+                url: url, status: status, error: error
+            });
+        });
+    }
+
+};
+
+var OAuthAddonSettingsViewModel = function(name, displayName) {
+    var self = this;
+    self.name = name;
+    self.properName = displayName;
+    self.accounts = ko.observableArray();
+    self.message = ko.observable('');
+    self.messageClass = ko.observable('');
+
+    self.setMessage = function(msg, cls) {
+        self.message(msg);
+        self.messageClass(cls || '');
```

Comment:
I pulled that code in from elsewhere, and honestly didn't think about it much. That makes sense.